v03_pipeline/lib/reference_datasets/misc.py

```diff
@@ -0,0 +1,31 @@
+import hail as hl
+
+from v03_pipeline.lib.model.dataset_type import DatasetType
+from v03_pipeline.lib.model.definitions import ReferenceGenome
+
+
+def enum_map(field: hl.Expression, enum_values: list[str]) -> dict:
+    lookup = hl.dict(
+        hl.enumerate(enum_values, index_first=False).extend(
+            # NB: adding missing values here allows us to
+            # hard fail if a mapped key is present and has an unexpected value
+            # but propagate missing values.
+            [(hl.missing(hl.tstr), hl.missing(hl.tint32))],
+        ),
+    )
+    if (
+        isinstance(field.dtype, hl.tarray | hl.tset)
+        and field.dtype.element_type == hl.tstr
+    ):
+        return field.map(
```

Comment:
Why did we originally have this notation `[f'{field_name}_ids']` in the enum map? Do we still need it?

Reply:
Ah.  I removed a lot of the logic that was building the `selects` and `enum_selects` dictionaries in favor of just selecting fields manually.  This means we'll need to name the fields ourselves... which I thought was actually clearer and more maintainable than the existing "magic", especially for `clinvar`, where we were doing both a select and then an enum select. 

For example, [assertion_ids](https://github.com/broadinstitute/seqr-loading-pipelines/pull/960/files#diff-caa8fea0ec85be082e2702ee31130dc4a8af587722e66be3ace335dd075198d0R153).

Reply:
except if you just changed the line for annotating the assertions in `select_fields` from `assertions=hl.if_else(` to `assertion=hl.if_else(` wouldn't the existing enum functionality properly name everything? Since seqr itself does "magic" parsing of the enums into fields, it feels like using shared functionality that assigns  field names to enums is much safer than every function needing to get this right

Reply:
it wasn't the `id` or `ids` suffix that was the issue, it was the two pass select... one to determine the field names and a second to actually do the transmute.  That didn't make sense to me if we're not configuring enums, but if we _are_, the calculus probably changes.  I can add it back.

Reply:
I think we are configuring them right? The current `select_fields` func does a select followed by a transmute so I feel like adding it back makes sense

Reply:
Cool, I added this back.  I'm still a little skeptical if the configuration is the right approach here in general (just because we have very few enum fields and rarely edit them), but this brings us to parity and is ok!